heron/common/src/java/com/twitter/heron/common/network/SocketChannelHelper.java

```diff
@@ -213,6 +214,10 @@ public void write() {
   // Force to flush all data in underneath buffer queue to socket with best effort
   // It is most likely happen when we are handling some unexpected cases, such as exiting
   public void forceFlushWithBestEffort() {
+    looper.exitLoop();
+    // Wait for NIO loop to confirm stopping process.
+    looper.waitForExit(10, TimeUnit.SECONDS);
```

Comment:
what happens if looper does not stop after 10 seconds

Reply:
The function stops waiting and returns false.

In this case, the code works pretty much the same way as before after the waiting. Except exitLoop() set a flag to true so the looper won't do any real work afterwards.